### PR TITLE
DM-37634: Update logBrowser to take ready-made butler

### PR DESCRIPTION
**Pull request by mfisherlevine**

### CODE REVIEW

File: python/lsst/summit/extras/logUtils.py [hunk 1]
```diff
@@ -21,7 +21,6 @@
 
 import logging
 import math
-import lsst.summit.utils.butlerUtils as butlerUtils
 
 __all__ = ["LogBrowser",
            ]
```
Comment (leeskelvin) [inline python/lsst/summit/extras/logUtils.py:26]: All on one line perhaps, i.e.:
```
__all__ = ["LogBrowser"]
```
(apologies for the out-of-code-change review)

File: python/lsst/summit/extras/logUtils.py [hunk 2]
```diff
@@ -35,6 +34,8 @@ class LogBrowser():
```
Comment (leeskelvin) [inline python/lsst/summit/extras/logUtils.py:33]: According to [the Python docs](https://docs.python.org/3/tutorial/classes.html#class-definition-syntax) (and my formatter), it looks like it's better to define this class without closing parentheses when empty, i.e.:
```
class LogBrowser:
```

Comment (mfisherlevine) [reply python/lsst/summit/extras/logUtils.py:33]: Tis a bad habit of mine from the bad old days. I just cleaned up a whole package the other day after Jim pointed that out 🙂

File: python/lsst/summit/extras/logUtils.py [hunk 3]
```diff
@@ -44,12 +44,13 @@ class LogBrowser():
...
-    amongst the ones you would like to be classed together, and add it to the
-    class property ``SPECIAL_ZOO_CASES`` to declare a new type of error animal.
+    like. If this is the case, take the first part of the message that is
+    common among the ones you would like to be classed together, and add it to
+    the class property ``SPECIAL_ZOO_CASES`` to declare a new type of error
+    animal.
 
     example usage:
-    logBrowser = LogBrowser(taskName=taskName, collection=collection)
+    logBrowser = LogBrowser(buter, taskName, collection)
```
Comment (leeskelvin) [inline python/lsst/summit/extras/logUtils.py:53]: `butler` typo.

Comment (mfisherlevine) [reply python/lsst/summit/extras/logUtils.py:53]: lol 🤦

File: python/lsst/summit/extras/logUtils.py [hunk 4]
```diff
@@ -44,12 +44,13 @@ class LogBrowser():
...
     function doesn't collapse them down to a single failure case as one would
-    like. If this is the case, the first part of the message that is common
-    amongst the ones you would like to be classed together, and add it to the
-    class property ``SPECIAL_ZOO_CASES`` to declare a new type of error animal.
+    like. If this is the case, take the first part of the message that is
+    common among the ones you would like to be classed together, and add it to
+    the class property ``SPECIAL_ZOO_CASES`` to declare a new type of error
+    animal.
 
     example usage:
```
Comment (leeskelvin) [inline python/lsst/summit/extras/logUtils.py:52]: You can put this into its own Numpydoc section, i.e.:
```
Examples
--------
```

Comment (mfisherlevine) [reply python/lsst/summit/extras/logUtils.py:52]: The reason I didn't want to do that is that sometimes those are actually considered literally runnable, and this one certainly isn't and could not really ever be, realistically, given the need for a butler and a collection.

File: python/lsst/summit/extras/logUtils.py [hunk 5]
```diff
@@ -61,12 +62,12 @@ class property ``SPECIAL_ZOO_CASES`` to declare a new type of error animal.
     SPECIAL_ZOO_CASES = ['with gufunc signature (n?,k),(k,m?)->(n?,m?)',
                          ]
```
Comment (leeskelvin) [inline python/lsst/summit/extras/logUtils.py:63]: Another recommended reformatting:
```
SPECIAL_ZOO_CASES = [
    "with gufunc signature (n?,k),(k,m?)->(n?,m?)",
]
```

Comment (leeskelvin) [reply python/lsst/summit/extras/logUtils.py:63]: Or actually, this is short enough for one line, apologies:
```
SPECIAL_ZOO_CASES = ["with gufunc signature (n?,k),(k,m?)->(n?,m?)"]
```

Comment (mfisherlevine) [reply python/lsst/summit/extras/logUtils.py]: Thanks. At some point this package will probably go `black` and that will be that.

**Comment by leeskelvin** [comment]
Thanks Merlin, interesting stuff! Apologies for going out of the bounds of the changes a couple of times! I restricted myself from recommending more file-wide changes (e.g., single quotes to double, one-line doc strings wrapped to a single line) - this looks good to me! :+1:

File: python/lsst/summit/extras/logUtils.py [hunk 6]
```diff
@@ -58,15 +59,16 @@ class property ``SPECIAL_ZOO_CASES`` to declare a new type of error animal.
         # butler.datastores is verbose by default and not interesting to most
         'lsst.daf.butler.datastores',
     ]
-    SPECIAL_ZOO_CASES = ['with gufunc signature (n?,k),(k,m?)->(n?,m?)',
-                         ]
+    SPECIAL_ZOO_CASES = [
+        'with gufunc signature (n?,k),(k,m?)->(n?,m?)',
+    ]
```
Comment (leeskelvin) [inline python/lsst/summit/extras/logUtils.py:64]: PS - I think my comment may have been lost in the noise, but you can wrap this onto one line if you wish (the trailing comma led me to put it as above, but one line works just fine).

File: python/lsst/summit/extras/logUtils.py (hunk 1 above)
Comment (leeskelvin) [reply python/lsst/summit/extras/logUtils.py]: Ah, fair enough, in which case I'd advocate using this format:
```
__all__ = [
    "LogBrowser",
]
```

**Review by leeskelvin — APPROVED** [review]

**Pull request merged**